crates/gpui/src/elements/anchored.rs

```diff
@@ -60,6 +60,12 @@ impl Anchored {
         self.fit_mode = AnchoredFitMode::SnapToWindow;
         self
     }
+
+    /// Snap to window edge and leave some margins.
+    pub fn snap_to_window_with_margin(mut self, edges: impl Into<Edges<Pixels>>) -> Self {
```

Comment:
I am not sure if this is the best name, I have considered changing the `snap to window` to add an argument, but sometimes the edge may not a must.